src/main/java/teammates/ui/webapi/StudentSearchIndexingWorkerAction.java

```diff
@@ -16,14 +17,27 @@ public ActionResult execute() {
         String courseId = getNonNullRequestParamValue(ParamsNames.COURSE_ID);
         String email = getNonNullRequestParamValue(ParamsNames.STUDENT_EMAIL);
 
-        StudentAttributes student = logic.getStudentForEmail(courseId, email);
-        try {
-            logic.putStudentDocument(student);
-        } catch (SearchServiceException e) {
-            // Set an arbitrary retry code outside of the range 200-299 to trigger automatic retry
-            return new JsonResult("Failure", HttpStatus.SC_BAD_GATEWAY);
+        if (isCourseMigrated(courseId)) {
+            Student student = sqlLogic.getStudentForEmail(courseId, email);
+            try {
+                sqlLogic.putStudentDocument(student);
+            } catch (SearchServiceException e) {
+                // Set an arbitrary retry code outside of the range 200-299 to trigger automatic retry
+                return new JsonResult("Failure", HttpStatus.SC_BAD_GATEWAY);
+            }
+    
+            return new JsonResult("Successful");
+        } else {
+            StudentAttributes student = logic.getStudentForEmail(courseId, email);
+            try {
+                logic.putStudentDocument(student);
+            } catch (SearchServiceException e) {
+                // Set an arbitrary retry code outside of the range 200-299 to trigger automatic retry
+                return new JsonResult("Failure", HttpStatus.SC_BAD_GATEWAY);
+            }
+    
+            return new JsonResult("Successful");
```

Comment:
hey @xenosf, while the logic here makes sense, it is slightly hard to read, let's change it, similar to how its done here: https://github.com/TEAMMATES/teammates/pull/12731

Reply:
Thanks for the feedback! Will refactor